dora/core/common/src/test/java/alluxio/membership/WorkerClusterViewTest.java

```diff
@@ -0,0 +1,194 @@
+    WorkerInfo worker1 = new WorkerInfo().setIdentity(id)
+        .setAddress(new WorkerNetAddress().setHost("host1"));
+    WorkerInfo worker2 = new WorkerInfo().setIdentity(id)
+        .setAddress(new WorkerNetAddress().setHost("host2"));
+    WorkerInfo worker3 = new WorkerInfo().setIdentity(id)
+        .setAddress(new WorkerNetAddress().setHost("host3"));
+    WorkerClusterView view = new WorkerClusterView(ImmutableList.of(worker1, worker2, worker3));
+    assertEquals(1, view.size());
+    assertEquals(Optional.of(id), view.getWorkerById(id).map(WorkerInfo::getIdentity));
+  }
+
+  @Test
+  public void iteration() {
+    WorkerIdentity worker1 = WorkerIdentityTestUtils.ofLegacyId(1);
+    WorkerIdentity worker2 = WorkerIdentityTestUtils.ofLegacyId(2);
+    Set<WorkerInfo> workers = ImmutableSet.of(
+        new WorkerInfo().setIdentity(worker1),
+        new WorkerInfo().setIdentity(worker2)
+    );
+    WorkerClusterView view = new WorkerClusterView(workers);
+
+    Set<WorkerInfo> workersFromView = new HashSet<>();
+    for (WorkerInfo w : view) {
+      workersFromView.add(w);
+    }
+    assertEquals(workers, workersFromView);
+  }
+
+  @Test
+  public void immutableIterator() {
+    WorkerClusterView view = new WorkerClusterView(ImmutableList.of(
+        new WorkerInfo().setIdentity(WorkerIdentityTestUtils.randomLegacyId())));
+    Iterator<WorkerInfo> iter = view.iterator();
+    assertTrue(iter.hasNext());
+    while (iter.hasNext()) {
+      iter.next();
+      assertThrows(UnsupportedOperationException.class, iter::remove);
+    }
+  }
+
+  @Test
+  public void size() {
+    WorkerIdentity worker1 = WorkerIdentityTestUtils.ofLegacyId(1);
+    WorkerIdentity worker2 = WorkerIdentityTestUtils.ofLegacyId(2);
+    List<WorkerInfo> workers = ImmutableList.of(
+        new WorkerInfo().setIdentity(worker1),
+        new WorkerInfo().setIdentity(worker2)
+    );
+    WorkerClusterView view = new WorkerClusterView(workers);
+    assertEquals(workers.size(), view.size());
+    assertFalse(view.isEmpty());
+  }
+
+  @Test
+  public void emptyView() {
+    WorkerClusterView view = new WorkerClusterView(ImmutableList.of());
+    assertEquals(0, view.size());
+    assertTrue(view.isEmpty());
+    assertEquals(ImmutableList.of(), view.stream().collect(Collectors.toList()));
+  }
+
+  @Test
+  public void equalityWorkersOrderDoesNotMatter() {
+    WorkerIdentity worker1 = WorkerIdentityTestUtils.ofLegacyId(1);
+    WorkerIdentity worker2 = WorkerIdentityTestUtils.ofLegacyId(2);
+    WorkerIdentity worker3 = WorkerIdentityTestUtils.ofLegacyId(3);
+    List<WorkerInfo> workers = ImmutableList.of(
+        new WorkerInfo().setIdentity(worker1),
+        new WorkerInfo().setIdentity(worker2),
+        new WorkerInfo().setIdentity(worker3)
+    );
+    Instant instant = Instant.now();
+    WorkerClusterView view1 = new WorkerClusterView(workers, instant);
+    WorkerClusterView view2 = new WorkerClusterView(Lists.reverse(workers), instant);
```

Comment:
There might be a tricky point here. `List.reverse()` is based on the same underlying list. So here the test passing doesn't guarantee two Views equal with different lists (lists have the same contents). In other words, a slightly better way
```
 WorkerClusterView view2 = new WorkerClusterView(new ArrayList<>(Lists.reverse(workers)), instant);
```